src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -3648,44 +3671,1767 @@ class B ''
+  IL_0056:  pop
+  IL_0057:  ldarg.1
+  IL_0058:  ldstr      "" = ""
+  IL_005d:  callvirt   ""System.Text.StringBuilder System.Text.StringBuilder.Append(string)""
+  IL_0062:  pop
+  IL_0063:  ldarg.1
+  IL_0064:  ldarg.0
+  IL_0065:  ldfld      ""int C1.B2""
+  IL_006a:  box        ""int""
+  IL_006f:  callvirt   ""System.Text.StringBuilder System.Text.StringBuilder.Append(object)""
+  IL_0074:  pop
+  IL_0075:  ldc.i4.1
+  IL_0076:  ret
+}
+");
+        }
+
+        [Fact]
+        public void ToString_DerivedRecord_AbstractSealed()
+        {
+            var src = @"
+record C1;
+abstract sealed record C2 : C1;
+";
+
+            var comp = CreateCompilation(src);
+            comp.VerifyEmitDiagnostics(
+                // (3,24): error CS0418: 'C2': an abstract type cannot be sealed or static
+                // abstract sealed record C2 : C1;
+                Diagnostic(ErrorCode.ERR_AbstractSealedStatic, "C2").WithArguments("C2").WithLocation(3, 24)
+                );
+
+            var print = comp.GetMember<MethodSymbol>("C2." + WellKnownMemberNames.PrintMembersMethodName);
+            Assert.Equal(Accessibility.Protected, print.DeclaredAccessibility);
+            Assert.True(print.IsOverride);
+            Assert.False(print.IsVirtual);
+            Assert.False(print.IsAbstract);
+            Assert.False(print.IsSealed);
+            Assert.True(print.IsImplicitlyDeclared);
+
+            var toString = comp.GetMember<MethodSymbol>("C1." + WellKnownMemberNames.ObjectToString);
+            Assert.Equal(Accessibility.Public, toString.DeclaredAccessibility);
+            Assert.True(toString.IsOverride);
+            Assert.False(toString.IsVirtual);
+            Assert.False(toString.IsAbstract);
+            Assert.False(toString.IsSealed);
+            Assert.True(toString.IsImplicitlyDeclared);
+        }
+
+        [Fact]
+        public void ToString_DerivedRecord_BaseHasSealedToString()
+        {
+            var src = @"
+record C1
+{
+    public sealed override string ToString() => throw null;
+}
+record C2 : C1;
+";
+
+            var comp = CreateCompilation(src);
+            comp.VerifyEmitDiagnostics(
+                // (4,35): error CS8870: 'C1.ToString()' cannot be sealed because containing record is not sealed.
+                //     public sealed override string ToString() => throw null;
+                Diagnostic(ErrorCode.ERR_SealedMethodInRecord, "ToString").WithArguments("C1.ToString()").WithLocation(4, 35),
+                // (6,8): error CS0239: 'C2.ToString()': cannot override inherited member 'C1.ToString()' because it is sealed
+                // record C2 : C1;
+                Diagnostic(ErrorCode.ERR_CantOverrideSealed, "C2").WithArguments("C2.ToString()", "C1.ToString()").WithLocation(6, 8)
+                );
+        }
+
+        [Fact]
+        public void ToString_DerivedRecord_TwoFieldsAndTwoProperties_ReverseOrder()
```

Comment:
I'm confused about what this test is demonstrating. It feels like it's just one of the previous tests with the names replaced--is the idea to show that lexical order is what matters, not some sorted order based on the names?

Reply:
> is the idea to show that lexical order is what matters

Exactly.